index.js

```diff
@@ -2,23 +2,29 @@
 
 const Promise = require('bluebird');
 const readInput = Promise.promisify(require('read'));
-const argv = require('yargs').argv;
+const argv = require('yargs').array('domain').argv;
 const xtend = require('xtend');
 const getCertificate = require('./lib.js');
 
-const read = (options) => () => {
+const read = (options) => (forceReadInput = false) => {
     const validator = options.validator ? options.validator : v => v && v.length > 0;
-    const argument = typeof argv[options.argument] === 'string' ? argv[options.argument] : undefined;
-    const resultPromise = argument ? Promise.resolve(argument) : readInput(options);
-    return resultPromise.then(value => {
-        if (validator(value)) {
+    const processValue = options.processValue ? options.processValue : v => v;
+    const resultPromise = forceReadInput ? readInput(options) : Promise.resolve(argv[options.argument]);
+    return resultPromise
+        .then(processValue)
+        .then(value => {
+            if (validator(value)) {
+                return value;
+            }
+
+            throw new Error();
```

Comment:
Please see my other [comment](https://github.com/rolodato/gitlab-letsencrypt/pull/14#discussion_r117643550). (The only reason I throw an error is to skip to `catch` code which starts a new iteration.)